client/src/pages/Register/FinalStep.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import Spinner from 'components/Spinner';
+import Template from 'components/Template';
+import PageWrapper from './PageWrapper';
+import debug from 'utils/debug';
+import { useRegFormData } from 'store/register/store';
+
+import './styles.scss';
+
+function FinalStep() {
+  const regFormData = useRegFormData();
+
+  debug('FinalStep data', regFormData);
+
+  const {
+    finalStep,
+    registration,
+    paymentInfo,
+    paymentStep,
+  } = regFormData;
+
+
+  if (
+    !paymentInfo ||
+    !registration ||
+    !paymentStep ||
+    !finalStep
+  ) return <Spinner />;
```

Comment:
Although this guard works pretty well, it may be better to redirect to `/form` if the `finalStep` data isn't present, what do you think?